lib/semian/net_http.rb

```diff
@@ -0,0 +1,90 @@
+require 'semian'
+require 'semian/adapter'
+
+module Net
+  ProtocolError.include(::Semian::AdapterError)
+
+  class SemianError < ::Net::ProtocolError
+    def initialize(semian_identifier, *args)
+      super(*args)
+      @semian_identifier = semian_identifier
+    end
+  end
+
+  ResourceBusyError = Class.new(SemianError)
+  CircuitOpenError = Class.new(SemianError)
+end
+
+module Semian
+  module NetHTTP
+    include Semian::Adapter
+
+    ResourceBusyError = ::Net::ResourceBusyError
+    CircuitOpenError = ::Net::CircuitOpenError
+
+    def semian_configuration
+      Semian::NetHTTP.retrieve_semian_configuration(address, port)
+    end
+
+    def semian_identifier
+      "nethttp_#{semian_configuration[:name]}"
+    end
+
+    DEFAULT_ERRORS = [
+      ::Timeout::Error, # includes ::Net::ReadTimeout and ::Net::OpenTimeout
+      ::TimeoutError, # alias for above
+      ::SocketError,
+      ::Net::HTTPBadResponse,
+      ::Net::HTTPHeaderSyntaxError,
+      ::Net::ProtocolError,
+      ::EOFError,
+      ::IOError,
+      ::SystemCallError, # includes ::Errno::EINVAL, ::Errno::ECONNRESET, ::Errno::ECONNREFUSED, ::Errno::ETIMEDOUT, and more
+    ].freeze # Net::HTTP can throw many different errors, this tries to capture most of them
+
+    class << self
+      attr_accessor :semian_configuration
+      attr_accessor :exceptions
+
+      def retrieve_semian_configuration(host, port)
+        @semian_configuration.call(host, port) if @semian_configuration.respond_to?(:call)
```

Comment:
They should probably read the instructions before using the adapter. That's the whole point of making things disable-until-explicitly-enabled - so that people who use it have to know what they're doing. 
Plus, I don't think it'll be an issue, since in order to figure out how to use this adapter they would have seen the example in the README, which shows (1) merging the `:name`, and (2) using `[:name]` it to form the `semian_identifier`.